src/czar/CzarRegistry.h

```diff
@@ -0,0 +1,127 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_CZAR_CZARREGISTRY_H
+#define LSST_QSERV_CZAR_CZARREGISTRY_H
+
+// System headers
+#include <atomic>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <thread>
+#include <unordered_map>
+#include <vector>
+
+// Third party headers
+#include "nlohmann/json.hpp"
+
+// Qserv headers
+#include "global/clock_defs.h"
+
+namespace lsst::qserv::cconfig {
+class CzarConfig;
+}  // namespace lsst::qserv::cconfig
+
+namespace lsst::qserv::czar {
+
+/// This class connects to the Replication System's Registry to register this czar and get
+/// worker contact information.
+/// The assumptions going forward are that the CzarChunkMap provides the real location of
+/// where all chunks are located and any workers in that map that are missing from this
+/// map are just temporary communications problems. A real prolonged failure of a worker
+/// will result in a new CzarChunkMap being created. As such, problems with missing
+/// worker contact information will be handled in Job creation
+/// in UserQueryFactory::newUserQuery and will be treated in similar manner as not being
+/// able to contact a worker.
+///
+/// There really shouldn't be communications problems, but there are, the best course of
+/// action would probably be to destroy the first instance of this and create a new one.
+///
+class CzarRegistry {
+public:
+    using Ptr = std::shared_ptr<CzarRegistry>;
+
+    /// Return a pointer to a new CzarRegistry object.
+    static Ptr create(std::shared_ptr<cconfig::CzarConfig> const& czarConfig) {
+        return Ptr(new CzarRegistry(czarConfig));
+    }
+
+    ~CzarRegistry();
+
+    struct WorkerContactInfo {
+        WorkerContactInfo(std::string const& wId_, std::string const& wHost_,
+                          std::string const& wManagementHost_, int wPort_, TIMEPOINT updateTime_)
+                : wId(wId_),
+                  wHost(wHost_),
+                  wManagementHost(wManagementHost_),
+                  wPort(wPort_),
+                  updateTime(updateTime_) {}
+        std::string const wId;              ///< key
+        std::string const wHost;            ///< "host-addr" entry.
+        std::string const wManagementHost;  ///< "management-host-name" entry.
+        int const wPort;                    ///< "management-port" entry.
+        TIMEPOINT const updateTime;         ///< "update-time-ms" entry.
+
+        /// Return true if all members, aside from updateTime, are equal.
+        bool same(WorkerContactInfo const& other) const {
```

Comment:
Why not define the usual operator?
```
bool operator==(WorkerContactInfo const& other) const;
```
Ignoring the timestamp from the comparison should be fine here since the timestamp is AKA "metadata" for the payload.